doc/src/manual/memory-management.md

```diff
@@ -0,0 +1,203 @@
+# [Memory Management and Garbage Collection](@id man-memory-management)
+
+Julia uses automatic memory management through its built-in garbage collector (GC). This section provides an overview of how Julia manages memory and how you can configure and optimize memory usage for your applications.
+
+## [Garbage Collection Overview](@id man-gc-overview)
+
+Julia features a garbage collector with the following characteristics:
+
+* **Non-moving**: Objects are not relocated in memory during garbage collection
+* **Generational**: Younger objects are collected more frequently than older ones
+* **Parallel and partially concurrent**: The GC can use multiple threads and run concurrently with your program
+* **Mostly precise**: The GC accurately identifies object references with minimal conservative scanning
+
+The garbage collector automatically reclaims memory used by objects that are no longer reachable from your program, freeing you from manual memory management in most cases.
+
+## [Memory Architecture](@id man-memory-architecture)
+
+Julia uses a two-tier allocation strategy:
+
+* **Small objects** (currently ≤ 2032 bytes but may change): Allocated using a fast per-thread pool allocator
```

Comment:
```suggestion
* **Small objects** (≤ 2032 bytes at the time of writing but may change): Allocated using a fast per-thread pool allocator
```

Reply:
This would give the impression it might not be correct/up to date. The docs should be the source of truth.

Reply:
I thought that it would be likely that it actually becomes out of date, but if you believe the opposite, feel free to discard this suggestion.